rskj-core/src/test/java/co/rsk/peg/NonStandardErpRedeemScriptBuilderFactoryTest.java

```diff
@@ -0,0 +1,86 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class NonStandardErpRedeemScriptBuilderFactoryTest {
+
+    ActivationConfig.ForBlock activations;
+    NetworkParameters networkParameters;
+
+    @BeforeEach
+    void setUp() {
+        activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
+    }
+
+    @Test
+    void preRSKIP284_returns_builderHardcoded_testnet() {
+        networkParameters = NetworkParameters.fromID(NetworkParameters.ID_TESTNET);
+
+        ErpRedeemScriptBuilder builder =
+            NonStandardErpRedeemScriptBuilderFactory.getNonStandardErpRedeemScriptBuilder(activations, networkParameters);
+
+        Assertions.assertTrue(builder instanceof NonStandardErpRedeemScriptBuilderHardcoded);
+    }
+
+    @Test
+    void preRSKIP284_doesnt_return_builderHardcoded_mainnet() {
+        networkParameters = NetworkParameters.fromID(NetworkParameters.ID_MAINNET);
+
+        ErpRedeemScriptBuilder builder =
+            NonStandardErpRedeemScriptBuilderFactory.getNonStandardErpRedeemScriptBuilder(activations, networkParameters);
+
+        Assertions.assertFalse(builder instanceof NonStandardErpRedeemScriptBuilderHardcoded);
+    }
+
+    @Test
```

Comment:
Consider using parameterized tests, at least for the ones that are the same but only the network changes